### PR TITLE
[docs] include upload logic in EAS migration doc

### PR DESCRIPTION
**Pull request by kbrandwijk**
# Why

ENG-4884

# How

Updated docs, linking to the existing fyi article for more details.

# Test Plan

NA

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build-reference/migrating.md [hunk 1]
```diff
@@ -30,6 +30,11 @@ Configuring the resizeMode or positioning of the splash screen with `splash` (or
 
 This often results in massive reductions in app size; managed apps built with EAS Build can be in the order of 10x smaller than the same app built with `expo build` ([learn why](https://blog.expo.dev/expo-managed-workflow-in-2021-5b887bbf7dbb)). The tradeoff here is that you need to be careful when publishing updates in order to avoid publishing an incompatible JavaScript bundle. Learn more in [updates](/build/updates.md).
 
+### Only files in your project folder that are not gitignored are uploaded to the build server
+
+Because EAS Build builds your full app, both the native part and the JavaScript part, on the server, it uses a different mechanism to upload your project files to the server than classic builds. Read more: ["How projects are uploaded to EAS Build
+"](https://expo.fyi/eas-build-archive).
```
Comment (brentvatne) [inline docs/pages/build-reference/migrating.md:36]:
```suggestion
 EAS Build builds your app like other CI services — in short, the entire project is uploaded securely to the cloud, then it is downloaded by a build server, the dependencies are installed, and the build is run ([learn more](https://docs.expo.dev/build-reference/ios-builds/)). Everything needed to build your app must be included in the project that is uploaded. The default mechanism for packaging your project is roughly equivalent to `git clone --depth 1`, and so anything that is in your `.gitignore` will not be uploaded (learn more in ["How projects are uploaded to EAS Build"](https://expo.fyi/eas-build-archive)).
 
 Developers often run into this with their "Google Services File", which they reference in their **app.json** / **app.config.js** but ignore in git. If anything in your project is gitignored but necessary for a successful build, you can either remove it from `.gitignore` and commit it, or [encode with base64 and store in EAS Secrets, then decode at build time](https://github.com/expo/fyi/blob/main/eas-build-archive.md#how-can-i-upload-files-to-eas-build-if-they-are-gitignored).
```

we should write up a

**Review by brentvatne — COMMENTED** [review]
i'm not crazy about the link in "encode with base64 and store in EAS Secrets, then decode at build time" -- it would be nice to have a dedicated docs page to creating files from secrets

**Comment by kbrandwijk** [comment]
> i'm not crazy about the link in "encode with base64 and store in EAS Secrets, then decode at build time" -- it would be nice to have a dedicated docs page to creating files from secrets

Sounds like a good follow up task.

**Review by brentvatne — APPROVED** [review]
approved but if someone can double check for grammar it would be appreciated, i'm a bit tired

**Review by Simek — COMMENTED** [review]
I have left few suggestions and questions, but as non-native speaker I'm not the best person to validate gramma. 😅

File: docs/pages/build-reference/migrating.md [hunk 2]
```diff
@@ -30,6 +30,12 @@ Configuring the resizeMode or positioning of the splash screen with `splash` (or
...
+
+ EAS Build builds your app like other CI services — in short, the entire project is uploaded securely to the cloud, then it is downloaded by a build server, the dependencies are installed, and the build is run ([learn more](https://docs.expo.dev/build-reference/ios-builds/)). Everything needed to build your app must be included in the project that is uploaded. The default mechanism for packaging your project is roughly equivalent to `git clone --depth 1`, and so anything that is in your `.gitignore` will not be uploaded (learn more in ["How projects are uploaded to EAS Build"](https://expo.fyi/eas-build-archive)).
+ 
+ Developers often run into this with their "Google Services File", which they reference in their **app.json** / **app.config.js** but ignore in git. If anything in your project is gitignored but necessary for a successful build, you can either remove it from `.gitignore` and commit it, or [encode with base64 and store in EAS Secrets, then decode at build time](https://github.com/expo/fyi/blob/main/eas-build-archive.md#how-can-i-upload-files-to-eas-build-if-they-are-gitignored).
```
[the remark this answers is not in the document]
Comment (kbrandwijk) [reply docs/pages/build-reference/migrating.md:37]: One is now 'ignored on Git', the other is 'ignored by Git'. I think I'm sticking with gitignored for both for now.

File: docs/pages/build-reference/migrating.md [hunk 3]
```diff
@@ -30,6 +30,12 @@ Configuring the resizeMode or positioning of the splash screen with `splash` (or
 
 This often results in massive reductions in app size; managed apps built with EAS Build can be in the order of 10x smaller than the same app built with `expo build` ([learn why](https://blog.expo.dev/expo-managed-workflow-in-2021-5b887bbf7dbb)). The tradeoff here is that you need to be careful when publishing updates in order to avoid publishing an incompatible JavaScript bundle. Learn more in [updates](/build/updates.md).
 
+### Only files in your project folder that are not gitignored are uploaded to the build server
```
[the remark this answers is not in the document]
Comment (kbrandwijk) [reply docs/pages/build-reference/migrating.md:33]:
```suggestion
### Only files in your project folder that are not ignored in git are uploaded to the build server
```

[the remark this answers is not in the document]
Comment (Simek) [reply docs/pages/build-reference/migrating.md:33]: "Ignored in Git" looks fine too, but please start with uppercase letter. 😉 
* https://git-scm.com/

**Review by Simek — APPROVED** [review]
Thank for the updates! :+1:

**Pull request merged**